### PR TITLE
_no_removeable_media rules update

### PR DESCRIPTION
**Pull request by freddieRv**
#### Description:

- Add OVAL checks to match the number of 'menuentry' entries to the number of 'set root' settings on grub.cfg for both the uefi and non-uefi rules 
- Also update the rule's yaml to mention this logic
- Update (as well as add) test scenarios for both the uefi and non-uefi _no_removeable_media rules to take the 'menuentry' entries into account

#### Rationale:

- The DISA STIG requirement that these rules implement states the following:
"_Check that the grub configuration file has the set root command in each menu entry_"
- Previously the grub.cfg file was searched for a single `set root` entry. This could lead to false-positives.

### CODE REVIEW

File: linux_os/guide/system/bootloader-grub2/uefi/uefi_no_removeable_media/rule.yml [hunk 1]
```diff
@@ -30,12 +30,14 @@ ocil_clause: 'it is not'
 
 ocil: |-
     To verify the system is not configured to use a boot loader on removable media,
-    run the following command:
+    check that the grub configuration file has the set root command in each menu
+    entry with the following commands:
+    <pre>$ sudo grep -cw menuentry {{{ grub2_uefi_boot_path }}}/grub.cfg</pre>
```
Comment (jan-cerny) [inline linux_os/guide/system/bootloader-grub2/uefi/uefi_no_removeable_media/rule.yml:35]: There should be mentioned that `grep -c` only prints a number which is count of the `menuentry` word occurrences.

**Review by jan-cerny — COMMENTED** [review]
It seems that the CI fail is caused by the fact that our CI runs in a container because when I execute the tests on a virtual machine backend I don't experience the problem:

```
[jcerny@thinkpad scap-security-guide{pr/9975}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel8 grub2_no_removeable_media,uefi_no_removeable_media
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-01-05-1335/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_grub2_no_removeable_media
INFO - Script no_such_line.fail.sh using profile (all) OK
INFO - Script cd.fail.sh using profile (all) OK
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script fd.fail.sh using profile (all) OK
INFO - Script multiple_correct_menu_entries.pass.sh using profile (all) OK
INFO - Script multiple_incorrect_menu_entries.fail.sh using profile (all) OK
INFO - Script multiple_menu_entries_missing_set_root.fail.sh using profile (all) OK
INFO - Script usb.fail.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_uefi_no_removeable_media
WARNING - Script hard_disk_set.pass.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script missing_hard_disk_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script removable_media_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
```

```
[jcerny@thinkpad scap-security-guide{pr/9975}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel8 --remediate-using ansible grub2_no_removeable_media,uefi_no_removeable_media
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-01-05-1350/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_grub2_no_removeable_media
INFO - Script no_such_line.fail.sh using profile (all) OK
INFO - Script cd.fail.sh using profile (all) OK
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script fd.fail.sh using profile (all) OK
INFO - Script multiple_correct_menu_entries.pass.sh using profile (all) OK
INFO - Script multiple_incorrect_menu_entries.fail.sh using profile (all) OK
INFO - Script multiple_menu_entries_missing_set_root.fail.sh using profile (all) OK
INFO - Script usb.fail.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_uefi_no_removeable_media
WARNING - Script hard_disk_set.pass.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script missing_hard_disk_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script removable_media_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
```

File: linux_os/guide/system/bootloader-grub2/uefi/uefi_no_removeable_media/rule.yml (hunk 1 above)
Comment (freddieRv) [reply linux_os/guide/system/bootloader-grub2/uefi/uefi_no_removeable_media/rule.yml:35]: Mentioned this on both rules after this line. Does it look good?

**Review by jan-cerny — APPROVED** [review]
```
[jcerny@thinkpad scap-security-guide{pr/9975}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel8 grub2_no_removeable_media,uefi_no_removeable_media
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-01-12-1309/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_grub2_no_removeable_media
INFO - Script no_such_line.fail.sh using profile (all) OK
INFO - Script cd.fail.sh using profile (all) OK
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script fd.fail.sh using profile (all) OK
INFO - Script multiple_correct_menu_entries.pass.sh using profile (all) OK
INFO - Script multiple_incorrect_menu_entries.fail.sh using profile (all) OK
INFO - Script multiple_menu_entries_missing_set_root.fail.sh using profile (all) OK
INFO - Script usb.fail.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_uefi_no_removeable_media
WARNING - Script hard_disk_set.pass.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script missing_hard_disk_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script removable_media_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
[jcerny@thinkpad scap-security-guide{pr/9975}]$ python3 tests/automatus.py rule --remediate-using ansible --libvirt qemu:///system ssgts_rhel8 grub2_no_removeable_media,uefi_no_removeable_media
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-01-12-1315/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_grub2_no_removeable_media
INFO - Script no_such_line.fail.sh using profile (all) OK
INFO - Script cd.fail.sh using profile (all) OK
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script fd.fail.sh using profile (all) OK
INFO - Script multiple_correct_menu_entries.pass.sh using profile (all) OK
INFO - Script multiple_incorrect_menu_entries.fail.sh using profile (all) OK
INFO - Script multiple_menu_entries_missing_set_root.fail.sh using profile (all) OK
INFO - Script usb.fail.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_uefi_no_removeable_media
WARNING - Script hard_disk_set.pass.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script missing_hard_disk_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
WARNING - Script removable_media_set.fail.sh using profile (all) notapplicable
WARNING - Rule xccdf_org.ssgproject.content_rule_uefi_no_removeable_media evaluation resulted in notapplicable
```

**Pull request merged**